src/poetry/utils/constants.py

```diff
@@ -3,3 +3,8 @@
 
 # Timeout for HTTP requests using the requests library.
 REQUESTS_TIMEOUT = 15
+
+RETRY_AFTER_HEADER = "retry-after"
+
+# Server response codes to retry requests on.
+STATUS_FORCELIST = [429, 500, 501, 502, 503, 504]
```

Comment:
Notes:
* I'm not sure about the 501, but it was already in the uploader code so kept it for compatibility
* Potentially we could add cloudflare 52x codes since it powers so much of the internet, even if it's not standard. See [Cloudflare docs](https://support.cloudflare.com/hc/en-us/articles/206973867-Status-code-metrics-in-Cloudflare-Site-Analytics#h_3VPW9Ekh0e580wUvfawvQv), [another example](https://github.com/python-gitlab/python-gitlab/blob/1020ce965ff0cd3bfc283d4f0ad40e41e4d1bcee/gitlab/client.py#L25)